### PR TITLE
SW-1158 high temperature module and fsm infrastructure

### PR DESCRIPTION
**Pull request by JFSolution**
- added fsm for high temperature warning
- refactor temperature manager
- added retriggering the cooling fan

### CODE REVIEW

**Comment by JFSolution** [comment]
Still missing are the unittests

File: octoprint_mrbeam/iobeam/temperature_manager.py [hunk 1]
```diff
@@ -253,64 +268,98 @@ def dismiss_high_temperature_warning(self):
...
+                and self.cooling_difference >= self.HYTERESIS_TEMPERATURE
+            ):
+                self._logger.warn(
+                    "Cooling break duration passed: %ss - Current temp: %s",
+                    self.cooling_duration,
+                    self.temperature,
+                )
+                self.cooling_resume()
+            else:
+                # self._logger.debug("Laser temperature nothing. Current temp: %s, self.is_cooling(): %s", self.temperatur, self.is_cooling())
```
Comment (caglarkilimci) [inline octoprint_mrbeam/iobeam/temperature_manager.py]: Do we need to keep it here?

File: tests/fsm/test_high_temperature_fsm.py [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+        "critical", "monitoring", "critically", False
+    )
+
+
+def test_transistion_from_warning_to_dismissed():
+    # Arrange
+    fsm = HighTemperatureFSM(MagicMock(), False, MagicMock())
+    fsm.start_monitoring()
+    fsm.warn()
+    assert fsm.warning.is_active
```
Comment (caglarkilimci) [inline tests/fsm/test_high_temperature_fsm.py:141]: Should this assert be here?

File: tests/fsm/test_high_temperature_fsm.py [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+    )
+
+
+def test_transistion_from_critical_to_dismissed():
+    # Arrange
+    fsm = HighTemperatureFSM(MagicMock(), False, MagicMock())
+    fsm.start_monitoring()
+    fsm.warn()
+    fsm.critical()
+    assert fsm.critically.is_active
```
Comment (caglarkilimci) [inline tests/fsm/test_high_temperature_fsm.py:176]: Should this assert be here? for all other tests as well.

**Comment by caglarkilimci** [comment]
Could you make clear for FSM? Link the ticket or a short doc in the code?

Could you please add doc strings for:
- HighTemperatureFSM Class
- temperature_manager.py functions

File: octoprint_mrbeam/analytics/analytics_handler.py [hunk 4]
```diff
@@ -679,6 +679,35 @@ def add_compressor_data(self, data, header_extension=None):
                 "Exception during add_compressor_static_data: {}".format(e)
             )
 
+    # High Temp Warning
+    def add_high_temp_warning_state_transition(
```
Comment (caglarkilimci) [inline octoprint_mrbeam/analytics/analytics_handler.py:683]: Docstring is missing.

File: octoprint_mrbeam/iobeam/temperature_manager.py [hunk 5]
```diff
@@ -121,33 +138,26 @@ def reset(self, kwargs):
...
-            if self.mode_time_based
-            else self.hysteresis_temperature,
+            key="cooling_duration",
+            value=self.cooling_duration,
         )
         self._logger.info(msg)
 
     @property
     def high_tmp_warn_threshold(self):
+        """Returns the temperature at which the user should be warned that the laserhead is too hot."""
```
Comment (caglarkilimci) [inline octoprint_mrbeam/iobeam/temperature_manager.py]: We can reformat to be consistent.

File: tests/fsm/test_high_temperature_fsm.py (hunk 2 above)
Comment (JFSolution) [reply tests/fsm/test_high_temperature_fsm.py:141]: Yes, we need it there.

**Review by caglarkilimci — APPROVED** [review]

**Pull request merged**